### PR TITLE
859 rename functions events

### LINKED ISSUE #859: [contracts] Rename functions + events
`prepare` / `fulfill` nomenclature doesn't really fit, and it's time for some renames:

- [x] `prepare` -> `xcall`
- [x] `fulfill` -> `execute`
- [x] `TransactionManager.sol` -> `Connext.sol`
- [x] `FulfillInterpreter.sol` -> `Executor.sol`
- [x] `transaction` -> `transfer`
- [x] `recipient` -> `to`

### PR DESCRIPTION
**Pull request by LayneHaber**
## The Problem

Closes #859 

## Checklist

- [ ] Test manually on `test-ui` using remote chains.
- [ ] Run load tests.
- [ ] Update documentation if needed.
- [ ] Update CHANGELOG.md.

### CODE REVIEW

File: packages/deployments/contracts/test/connext.spec.ts [hunk 1]
```diff
@@ -887,37 +886,36 @@ describe.only("TransactionManager", () => {
...
+        to: user.address,
         callData: "0x",
         originDomain: `${originDomain}`,
         destinationDomain: `${destinationDomain}`,
       };
       const transactingAssetId = originAssets[assetIdx].address;
       const amount = 1000;
       // check balance of user
-      const prepare = await originTm.connect(user).prepare({ params, transactingAssetId, amount });
+      const prepare = await originTm.connect(user).xcall({ params, transactingAssetId, amount });
```
Comment (jakekidd) [inline packages/deployments/contracts/test/connext.spec.ts:897]: nit: Doesn't `originTm` stand for `originTransactionmanager` ? lol

Comment (jakekidd) [reply packages/deployments/contracts/test/connext.spec.ts:897]: nvm this is just a unit test, ignore thi

File: packages/deployments/contracts/typechain-types/Executor.ts [hunk 2]
```diff
@@ -17,42 +17,38 @@ import { FunctionFragment, Result, EventFragment } from "@ethersproject/abi";
 import { Listener, Provider } from "@ethersproject/providers";
 import { TypedEventFilter, TypedEvent, TypedListener, OnEvent } from "./common";
 
-export interface FulfillInterpreterInterface extends utils.Interface {
-  contractName: "FulfillInterpreter";
+export interface ExecutorInterface extends utils.Interface {
```
Comment (jakekidd) [inline packages/deployments/contracts/typechain-types/Executor.ts:20]: I like this name better!

File: packages/deployments/contracts/typechain-types/Executor.ts [hunk 3]
```diff
@@ -17,42 +17,38 @@ import { FunctionFragment, Result, EventFragment } from "@ethersproject/abi";
...
 export type ExecutedEvent = TypedEvent<
-  [string, string, string, string, BigNumber, string, string, boolean, boolean],
+  [string, string, string, BigNumber, string, string, boolean, boolean],
   {
-    transactionId: string;
-    callTo: string;
+    transferId: string;
+    to: string;
     assetId: string;
-    fallbackAddress: string;
```
Comment (jakekidd) [inline packages/deployments/contracts/typechain-types/Executor.ts:55]: no more fallbackAddress ?

File: packages/deployments/contracts/contracts/Connext.sol [hunk 4]
```diff
@@ -337,15 +336,15 @@ contract TransactionManager is Initializable, ReentrancyGuardUpgradeable, Propos
   IWrapped public wrapper;
 
   /**
-   * @notice Nonce for the contract, used to keep unique transaction ids.
-   * @dev Assigned at first interaction (prepare on origin domain);
+   * @notice Nonce for the contract, used to keep unique transfer ids.
+   * @dev Assigned at first interaction (xcall on origin domain);
    */
-  uint256 public nonce ;
```
Comment (jakekidd) [inline packages/deployments/contracts/contracts/Connext.sol:343]: finally that whitespace is gone, it was killing me

File: packages/deployments/contracts/contracts/Connext.sol [hunk 5]
```diff
@@ -451,9 +452,10 @@ contract TransactionManager is Initializable, ReentrancyGuardUpgradeable, Propos
     nonce = 0;
     domain = _domain;
     bridgeRouter = BridgeRouter(_bridgeRouter);
-    interpreter = new FulfillInterpreter(address(this));
+    executor = new Executor(address(this));
     tokenRegistry = TokenRegistry(_tokenRegistry);
     wrapper = IWrapped(_wrappedNative);
+    EMPTY = hex"c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470";
```
Comment (jakekidd) [inline packages/deployments/contracts/contracts/Connext.sol:458]: what is this EMPTY?

File: packages/deployments/contracts/contracts/interfaces/IConnext.sol [hunk 6]
```diff
@@ -0,0 +1,6 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.11;
+
+interface IConnext {
+
```
Comment (jakekidd) [inline packages/deployments/contracts/contracts/interfaces/IConnext.sol:5]: great interface, much useful 10/10

File: packages/deployments/contracts/contracts/interpreters/Executor.sol [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+    // Emit event
+    emit Executed(
+      transferId,
+      to,
+      assetId,
+      amount,
+      callData,
+      returnData,
+      success,
+      isContract
```
Comment (jakekidd) [inline packages/deployments/contracts/contracts/interpreters/Executor.sol:99]: Why do we need `isContract` anymore? There's no Router.sol anymore so

**Review by jakekidd — APPROVED** [review]
left 1 (legit) comment, otherwise lgtm

File: packages/deployments/contracts/contracts/Connext.sol [hunk 8]
```diff
@@ -33,7 +31,7 @@ import "@openzeppelin/contracts-upgradeable/token/ERC20/utils/SafeERC20Upgradeab
 // 1. Finalize BridgeMessage / BridgeRouter structure + backwards compatbility
 // 2. Gas optimizations
 
-contract TransactionManager is Initializable, ReentrancyGuardUpgradeable, ProposedOwnableUpgradeable, MerkleTreeManager {
+contract Connext is Initializable, ReentrancyGuardUpgradeable, ProposedOwnableUpgradeable, MerkleTreeManager {
 
   // TODO: why is this breaking the build
   // uint256 internal constant 3 = 3;
```
Comment (jakekidd) [inline packages/deployments/contracts/contracts/Connext.sol:37]: why *is* this breaking the build?? lol

maybe bc u (or whoever wrote) named the constant `3` ?

File: packages/deployments/contracts/contracts/interpreters/Executor.sol (hunk 7 above)
Comment (sanchaymittal) [reply packages/deployments/contracts/contracts/interpreters/Executor.sol:99]: Here, isContract is for metrics only. Keeps the track of callTo data execution.

**Review by sanchaymittal — APPROVED** [review]
looks good

**Pull request merged**